src/libcore/num/mod.rs

```diff
@@ -1471,3 +1472,73 @@ impl fmt::Display for ParseIntError {
 }
 
 pub use num::dec2flt::ParseFloatError;
+
+
+// Conversion traits for primitive integer types
+// Conversions T -> T are covered by a blanket impl and therefore excluded
+macro_rules! impl_from {
+    ($Small: ty, $Large: ty) => {
+        impl From<$Small> for $Large {
+            #[stable(feature = "lossless_int_conv", since = "1.5.0")]
+            #[inline]
+            fn from(small: $Small) -> $Large {
+                small as $Large
+            }
+        }
+    }
+}
+
+// Unsigned -> Unsigned
+impl_from! { u8, u16 }
+impl_from! { u8, u32 }
+impl_from! { u8, u64 }
+impl_from! { u8, usize }
+impl_from! { u16, u32 }
+impl_from! { u16, u64 }
+impl_from! { u16, usize }
+impl_from! { u32, u64 }
+#[cfg(any(target_pointer_width = "64", target_pointer_width = "32"))]
+impl_from! { u32, usize }
+#[cfg(target_pointer_width = "64")]
```

Comment:
> Well, currently all this code that would fail to compile with these changes is silently truncating the integers.

No. The compiler is truncating them because the user explicitly asked it to. This RFC will get rid of unnecessary `a as b` conversions allowing the programmer to pay more attention to the important ones (the ones that might truncate). I write all my code on a 64bit machine and I *don't* want to have to make a second pass fixing all of my conversions when someone comes along with a 32bit machine. I'd much rather spend the time reasoning about conversion safety up front.